packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts

```diff
@@ -292,13 +294,14 @@ export class SummarizerNode implements IRootSummarizerNode {
         }
     }
 
-    private refreshLatestSummaryFromSnapshot(
+    protected async refreshLatestSummaryFromSnapshot(
```

Comment:
I have not traced code, but please note that any time we add async processing to something that is tied to processes that we do not control directly, we are introducing potential for reentrnacy.
I.e. can we get into this function with next ack while previous is still processing and did not finish? If yes, this code more likely than not would not handle it correctly. Processing of "future" ack may finish before processing of prior "ack" is done, or happen in parallel

Reply:
+ @arinwt to also take a look at if this scenario is possible.
I was a little worried about the async nature of the whole summary process too and I discussed this with Arin. I think we need to be more assertive about some of the assumptions we have around summary, similar to what you had mentioned to me earlier about asserting that there are no inbound messages during summary.

Reply:
We can handle reentrancy here by checking that the referenceSequenceNumber we have already seen is greater than the one we are processing. If its not, we can simply return because it means we have seen a later summary ack and updated our state so this one can be ignored.

Reply:
I will think about testing this and adding test cases for it.

Reply:
Right this is exactly what we discussed.  I do want to point out that any async issues would likely not be new to this PR, since the calling function is already async: `async refreshLatestSummary` calls `refreshLatestSummaryFromSnapshot`.

That being said, I think re-entrancy is unlikely just by nature of summaries, but would be good to add some promise/locking to make sure it is not possible.

Additional async scenario we could prevent (not sure if it is actually possible in practice), is beginning to summarize before finishing the call to refreshLatestSummaryAck.

Reply:
Thanks Arin. I added check for referecenceSequenceNumber here. Can you check if it looks fine.

Reply:
I see the check, and I think this will correctly prevent re-entrancy (like we discussed), and in the future we can address the issue I mentioned above using a lock+promise or something similar.